workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -476,5 +450,91 @@ private static IEnumerable<string> FindDirInPackages(string directory)
 
             return packagePaths.Union(cachedPackagePaths).Distinct();
         }
+
+        internal static void GenerateCodegenRunConfigs(GdkToolsConfiguration toolsConfig = null)
+        {
+            toolsConfig = toolsConfig ?? GdkToolsConfiguration.GetOrCreateInstance();
+
+            var schemaCompilerPath = GetSchemaCompilerPath();
+            var logfilePath = GetDefaultLoggerOutputPath();
+
+            var codegenArgs = new List<string>
+            {
+                $"--json-dir=\"{ImprobableJsonDir}\"",
+                $"--schema-compiler-path=\"{schemaCompilerPath}\"",
+                $"--worker-json-dir=\"{WorkerJsonPath}\"",
+                $"--log-file=\"{logfilePath}\""
+            };
+
+            if (toolsConfig.VerboseLogging)
+            {
+                codegenArgs.Add($"--verbose");
+            }
+
+            var codegenOutputPath = Path.GetFullPath(Path.Combine(Application.dataPath, "..", toolsConfig.CodegenOutputDir));
+            codegenArgs.Add($"--native-output-dir=\"{codegenOutputPath}\"");
+
+            // Add user defined schema directories
+            codegenArgs.AddRange(toolsConfig.SchemaSourceDirs
```

Comment:
If an input directory doesn't exist, does that imply the user has something mis-configured? It may be better to at least warn that a directory doesn't exist

Reply:
mis-configuration is captured when we validate the tools config, but we should still warn here in case a directory a user expects is not present for whatever reason